src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -647,12 +650,27 @@ private ImmutableArray<ParameterSymbol> ComputeParameters(DiagnosticBag diagnost
                     diagnostics.Add(ErrorCode.ERR_ParameterIsStaticClass, this.locations[0], propertyType.Type);
                 }
 
-                parameters.Add(new SynthesizedAccessorValueParameterSymbol(this, propertyType, parameters.Count));
+                parameters.Add(new SynthesizedAccessorValueParameterSymbol(this, propertyType, parameters.Count, _property.SetterValueFlowAnalysisAnnotations));
             }
 
             return parameters.ToImmutableAndFree();
         }
 
+        internal override void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+        {
+            base.AddSynthesizedReturnTypeAttributes(moduleBuilder, ref attributes);
+
+            var compilation = this.DeclaringCompilation;
+            if ((ReturnTypeFlowAnalysisAnnotations & FlowAnalysisAnnotations.MaybeNull) != 0)
+            {
+                AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_Diagnostics_CodeAnalysis_MaybeNullAttribute__ctor));
```

Comment:
>AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_Diagnostics_CodeAnalysis_MaybeNullAttribute__ctor)); [](start = 16, length = 150)

We shouldn't be looking up for a type and potentially finding a different type by comparison to the type used by the user, or failing to find it due to an ambiguity. We should simply apply an attribute from the property.

Reply:
I'll stop by to clarify

---
In reply to: [295420589](https://github.com/dotnet/roslyn/pull/36444#discussion_r295420589) [](ancestors = 295420589)